src/activities/Elsa.Activities.Http/Activities/SendHttpRequest/SendHttpRequest.cs

```diff
@@ -257,7 +259,20 @@ private HttpRequestMessage CreateRequest()
 
             return request;
         }
-        
+
+        private async Task<string?> GetAuthorizationHeader() {
+            if (_authorizationHeaderHandlers.Count == 0 || string.IsNullOrEmpty(Authorization)) {
+                return Authorization;
+            }
+
+            var header = Authorization;
+            foreach (var handler in _authorizationHeaderHandlers) {
+                header = await handler.EvaluateStoredValue(header);
```

Comment:
1. it could be and could be not. The main idea is to decouple the dependency and keep it vague enough. Potentially you can have chained handlers (e.g. one handler replacing/injecting some specific part of a header, second one replacing or appending something else). Secret handler treats a header ultimately as its own when it has specific prefix.
2. Initially I was trying to proceed with ExpressionHandler but it caused quite some dependency changes - starting from ExpressionEvaluator further on (there could have been some issue with lifecycles as well). In addition a new syntax would be needed and it would be quite specific to this module (Syntaxes are currently defined quite statically so I would need to add it to core project) - as I understood from Sipkes previous comment this module should be completely separate and not in the core concepts. If Sipke can comment on what strategy should there be for introducing additional syntaxes. Not sure if having "unknown syntaxes" is the best here.

Reply:
@sfmskywalker what are your thoughts on this?

Reply:
@sfmskywalker it totally makes sense, though it feels like an its own story. Is it ok to add a static Secret Syntax for this particular task and then make Syntaxes more configurable (i.e. every module can add various syntaxes dynamically)?

Reply:
@sfmskywalker pushed a version with Syntax used instead of designated handlers. Please have a look

Reply:
Thanks @ke-real, I'll take a look soon!

Reply:
@ke-real This looks good! I did have one question about the following UI:

<img width="990" alt="image" src="https://user-images.githubusercontent.com/938393/207958694-ccde5812-4440-47f6-9949-a11d8c7bb51e.png">

Is it necessary to display the "Secret" syntax from here? If not, I suggest we remove it, since the "Default" UI is already replaced server-side using `DescribingActivityTypeHandler` and the enrichers it invokes - which set the default syntax to "Secret".

Other than that, I think it's good to go 👍🏻

Reply:
@sfmskywalker  Removed explicit Secret syntax, works fine. 

Also changed secret referencing by Name as proposed in #3520 - it is also useful for our purposes when it is needed to replicate a workflow setup in several environments